File: java/0746-min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int one = 0;
        int two = 0;

        for (int i = cost.length - 1; i >= 0; i--) {
            cost[i] += Math.min(one, two);
            two = one;
            one = cost[i];
        }

        return Math.min(cost[0], cost[1]);
    }
}

File: kotlin/0746-min-cost-climbing-stairs.kt
class Solution {
    fun minCostClimbingStairs(cost: IntArray): Int {
        for (i in cost.size - 3 downTo 0) {
            cost[i] += minOf(cost[i + 1], cost[i + 2])
        }
        return minOf(cost[0], cost[1])
    }
}